app/components/UI/Stake/hooks/useStakingInput.ts

```diff
@@ -0,0 +1,160 @@
+import { BN } from 'ethereumjs-util';
+import { useState, useMemo, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  selectCurrentCurrency,
+  selectConversionRate,
+} from '../../../../selectors/currencyRateController';
+import {
+  toWei,
+  weiToFiatNumber,
+  renderFromTokenMinimalUnit,
+  fiatNumberToWei,
+  fromTokenMinimalUnitString,
+  limitToMaximumDecimalPlaces,
+  renderFiat,
+} from '../../../../util/number';
+
+const useStakingInputHandlers = (balance: BN) => {
+  const [amount, setAmount] = useState('0');
+  const [amountBN, setAmountBN] = useState<BN>(new BN(0));
+  const [estimatedAnnualRewards, setEstimatedAnnualRewards] = useState('-');
+
+  const isNonZeroAmount = useMemo(() => amountBN.gt(new BN(0)), [amountBN]);
+  const isOverMaximum = useMemo(() => {
+    const additionalFundsRequired = amountBN.sub(balance || new BN(0));
+    return isNonZeroAmount && additionalFundsRequired.gt(new BN(0));
+  }, [amountBN, balance, isNonZeroAmount]);
+
+  const [fiatAmount, setFiatAmount] = useState('0');
+  const [isEth, setIsEth] = useState<boolean>(true);
+  const currentCurrency = useSelector(selectCurrentCurrency);
+  const conversionRate = useSelector(selectConversionRate) || 1;
+
+  const annualRewardRate = '0.026'; //TODO: Replace with actual value: STAKE-806
+
+  const currencyToggleValue = isEth
+    ? `${fiatAmount} ${currentCurrency.toUpperCase()}`
+    : `${amount} ETH`;
+
+  const handleEthInput = useCallback(
+    (value: string) => {
+      setAmount(value);
+      setAmountBN(toWei(value, 'ether'));
+      const fiatValue = weiToFiatNumber(
+        toWei(value, 'ether'),
+        conversionRate,
+        2,
+      ).toString();
+      setFiatAmount(fiatValue);
+    },
+    [conversionRate],
+  );
+
+  const handleFiatInput = useCallback(
+    (value: string) => {
+      setFiatAmount(value);
+      const ethValue = renderFromTokenMinimalUnit(
+        fiatNumberToWei(value, conversionRate).toString(),
+        18,
+        5,
+      );
+
+      setAmount(ethValue);
+      setAmountBN(toWei(ethValue, 'ether'));
+    },
+    [conversionRate],
+  );
+
+  /* Keypad Handlers */
+  const handleKeypadChange = useCallback(
+    ({ value }) => {
+      isEth ? handleEthInput(value) : handleFiatInput(value);
+    },
+    [handleEthInput, handleFiatInput, isEth],
+  );
+
+  const handleCurrencySwitch = useCallback(() => {
+    setIsEth(!isEth);
+  }, [isEth]);
+
+  const percentageOptions = [
+    { value: 0.25, label: '25%' },
+    { value: 0.5, label: '50%' },
+    { value: 0.75, label: '75%' },
+    { value: 1, label: 'Max' },
```

Comment:
Should we break `Max` out into `en.json` for translation support?
```suggestion
    { value: 1, label: strings('stake.max') },
```